atlasdb-client/src/main/java/com/palantir/atlasdb/table/description/IndexDefinition.java

```diff
@@ -211,18 +173,21 @@ public IndexType getIndexType() {
         return indexType;
     }
 
+    public void validate() {
```

Comment:
I normally find it simpler to have a chunk of ways to early return, then normal running of the code - makes it more likely for people to add extra steps in the initial rather than nest the ifs and conditions more and more.

In this case:

```
if (ignoreHotspottingChecks) {
    return;
}

Preconditions.check...
Preconditions.check...
```